python/hsfs/util.py

```diff
@@ -143,12 +142,21 @@ def get_hudi_datestr_from_timestamp(timestamp):
 def convert_event_time_to_timestamp(event_time, time_zone=timezone.utc):
     if event_time is None:
         return None
-    if isinstance(event_time, str):
+    elif isinstance(event_time, str):
         if event_time:
             return get_timestamp_from_date_string(event_time, time_zone)
-    if isinstance(event_time, pd._libs.tslibs.timestamps.Timestamp):
-        # pandas.timestamp represents millisecond in decimal
+    elif isinstance(event_time, pd._libs.tslibs.timestamps.Timestamp):
+        # convert to unix epoch time in milliseconds.
```

Comment:
`pd.Timestamps` can also have timezone information. To keep the logic consistent between pd.Timestamp and `datetime.datetime` you can first convert the pd.Timestamp to a datetime object by calling `event_time.to_pydatetime()` (`https://pandas.pydata.org/docs/reference/api/pandas.Timestamp.to_pydatetime.html`) and then re-use the datetime logic.

Reply:
I'd also be fine with dropping pd.Timezone support. I see that in the typehints it's also not mentioned and maybe it doesn't add much value, since users can call .to_pydatetime() on pd.Timestamp object if required.

Reply:
done